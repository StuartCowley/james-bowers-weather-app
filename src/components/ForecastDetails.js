import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastDetails.css";
import WeatherIcon from "react-icons-weather";

function ForecastDetails({ forecast }) {
  const { date, humidity, icon, temperature, wind } = forecast;
  const formattedDate = new Date(date).toDateString().slice(0, 10);
  return (
    <div className="forecast-details__wrapper">
      <div className="forecast-details__left">
        <div className="forecast-details__icon">
          <WeatherIcon name="owm" iconId={icon} />
        </div>
        <div
          className="forecast-details__date"
          data-testid="forecast-details__date"
        >
          {formattedDate}
        </div>
      </div>
      <div className="forecast-details__right">
        <div className="forecast-details__temperature">
          <p className="forecast-details__heading">Temperature</p>
          <p>Max: {temperature.max}°C</p>
          <p>Min: {temperature.min}°C</p>
        </div>
        <div className="forecast-details__humidity">
          <p>Humidity: {humidity}%</p>
        </div>
        <div className="forecast-details__wind">
          <p className="forecast-details__heading">Wind</p>
          <div className="forecast-details__wind-speed">
            <p>Speed: {wind.speed}m/s</p>
          </div>
          <div className="forecast-details__wind-direction">
            <p>Direction: {wind.direction.toUpperCase()}</p>
          </div>
        </div>
      </div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    icon: PropTypes.string.isRequired,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
    humidity: PropTypes.number.isRequired,
  }).isRequired,
};
