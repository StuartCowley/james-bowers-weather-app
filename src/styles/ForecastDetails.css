.forecast-details__wrapper {
    display: grid;
    flex-direction: row;
    grid-template-columns: 1fr 1fr;
    height: 100%;
    outline: 2px solid white;
    padding: 6px;
    margin: 8px;
    text-align: center;
    color: white;
    font-weight: 400;
    background-color: rgba(255, 255, 255, 0.2);
    -moz-box-shadow:    inset 0 0 20px rgba(0, 0, 0, 0.2);
    -webkit-box-shadow: inset 0 0 20px rgba(0, 0, 0, 0.2);
    box-shadow:         inset 0 0 20px rgba(0, 0, 0, 0.2),
                              -10px 10px 20px rgba(0,0,0,0.2);
    text-shadow: 0 0 5px rgba(0, 0, 0, 0.2);
    flex-grow: 5;
    min-width: 300px;
}
.forecast-details__left {
    display: flex;
    flex-direction: column;
    align-items: right;
    /* justify-content: center; */
    text-align: right;
    margin-right: 3%;
}
.forecast-details__right {
    display: flex;
    flex-direction: column;
    text-align: left;
    margin-left: 3%;
}

.forecast-details__heading {
    font-size: 1.5rem;
    margin-top: 8px;
    font-weight: 400;
    text-shadow: 0 0 5px rgba(0, 0, 0, 0.3);
}
p {
   margin: 0 0 8px 0; 
}
.forecast-details__icon{
    font-size: 6rem;
    line-height: 6rem;
    height: 6rem;
    color: white;
    text-shadow: 0 0 20px rgba(0, 0, 0, 0.5);
    margin: 2rem 0;
    text-align: right;
}
.forecast-details__date {
    display: flex;
    justify-content: right;
    font-size: 2rem;
    font-weight: 500;
    padding: 0.2rem 0;
    text-align: center;
    margin-right: 0;
    /* border-bottom: 2px solid white; */
}