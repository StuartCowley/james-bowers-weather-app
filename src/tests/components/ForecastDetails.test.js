import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = [
    {
      date: 1111111111111,
      temperature: {
        max: 25,
        min: 17,
      },
      humidity: 20,
      wind: {
        speed: 5,
        direction: "W",
      },
    },
    {
      date: 2222222222222,
      temperature: {
        max: 13,
        min: 5,
      },
      humidity: 80,
      wind: {
        speed: 30,
        direction: "E",
      },
    },
  ];
  function formatDate(date) {
    return new Date(date).toDateString().slice(0, 10);
  }

  it("renders correctly", () => {
    const { asFragment, getByText } = render(
      <ForecastDetails forecast={validProps[0]} />
    );

    expect(asFragment()).toMatchSnapshot();

    expect(getByText(formatDate(validProps[0].date))).toHaveAttribute(
      "class",
      "forecast-details__date"
    );
  });
});
