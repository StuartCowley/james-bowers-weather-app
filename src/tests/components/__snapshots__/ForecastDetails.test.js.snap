// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
      data-testid="forecast-details__date"
    >
      Fri Mar 18
    </div>
    <div
      class="forecast-details__temperature"
    >
      <p
        class="forecast-details__heading"
      >
        Temperature
      </p>
      <p>
        Max: 25°C
      </p>
      <p>
        Min: 17°C
      </p>
    </div>
    <div
      class="forecast-details__humidity"
    >
      <p>
        Humidity: 20%
      </p>
    </div>
    <div
      class="forecast-details__wind"
    >
      <p
        class="forecast-details__heading"
      >
        Wind
      </p>
      <div
        class="forecast-details__wind-speed"
      >
        <p>
          Speed: 5m/s
        </p>
      </div>
      <div
        class="forecast-details__wind-direction"
      >
        <p>
          Direction: W
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
